final

Nikolya777777777777 opened the pull request.

mateacademy-ai-mentor commented on src/main/java/mate/academy/service/AuthenticationServiceImpl.java:
```diff
@@ -2,21 +2,24 @@
+        logger.debug("Method \'login\' was called. Params: login={}", login);
         User user = findByLogin(login);
         if (!user.getPassword().equals(password)) {
-            throw new AuthenticationException("Username or password are incorrect");
+            logger.error("Username or password are incorrect");
         }
```
The `login` method should throw an `AuthenticationException` when the password is incorrect, as indicated by the method signature. Consider adding a throw statement here.

mateacademy-ai-mentor reviewed (changes requested):
Thank you for your hard work on this pull request! 😊 While most of the code is well-structured and meets the task requirements, there is a critical issue that needs to be addressed before we can approve it. The `login` method in the `AuthenticationServiceImpl` class should throw an `AuthenticationException` when the password is incorrect, as indicated by the method signature. This is crucial for the functionality and security of the authentication process. Once this is resolved, your solution will be ready for approval. Keep up the great work, and I look forward to your updated submission! 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job on addressing the previous issues and aligning your code with the task requirements! 🎉

Your implementation of the logging mechanism using Log4j in both `Main.java` and `OrderServiceImpl.java` is well-structured and meets the guidelines. The `.gitignore` file is correctly set up to ignore log files, ensuring that unnecessary files are not pushed to GitHub. It's also commendable that you've replaced `System.out.println()` and `e.printStackTrace()` with logger methods, which is a best practice for maintaining clean and professional code.

The `AuthenticationServiceImpl.java` file now correctly throws an `AuthenticationException` when the password is incorrect, which was a critical issue in the previous review. This change is crucial for the functionality of your authentication system.

While there are no critical issues that need immediate attention, always remember to review any comments for potential improvements. This will help you refine your skills and produce even better code in the future. Keep up the great work, and continue learning from each experience! 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

okuzan reviewed (changes requested):
?? You deployed bookshop project in jv-logger PR?

Nikolya777777777777 commented:
I answered to your question in jv-chat, please check, I didn't import anything, all these files were there at once, I changed pom.xml, added the log4j2.xml file and created loggers in the classes, added logger.debug, error and info. If I understood your question right)

Elena-Bruyako reviewed (changes requested):
Task implemented correctly, but why there are so many files? revert changes in gitignore file
